pkg/asset/installconfig/aws/validation.go

```diff
@@ -369,10 +404,28 @@ func validateExistingHostedZone(session *session.Session, ic *types.InstallConfi
 			fmt.Sprintf("hosted zone domain %q is not a parent of the cluster domain %q", *zone.HostedZone.Name, dottedClusterDomain)))
 	}
 
+	problematicRecords, err := getProblematicRecords(client, zone.HostedZone, dottedClusterDomain)
+	if err != nil {
+		allErrs = append(allErrs, field.InternalError(hostedZonePath,
+			errors.Wrapf(err, "could not list record sets for hosted zone %q", ic.AWS.HostedZone)))
+
+	}
+
+	if len(problematicRecords) > 0 {
+		detail := fmt.Sprintf(
+			"hosted zone already has record sets for the domain of the cluster: [%s]",
+			strings.Join(problematicRecords, ", "),
+		)
+		allErrs = append(allErrs, field.Invalid(hostedZonePath, ic.AWS.HostedZone, detail))
+	}
+
+	return allErrs
+}
+
+func getProblematicRecords(client *route53.Route53, hostedZone *route53.HostedZone, dottedClusterDomain string) (problematicRecords []string, err error) {
```

Comment:
This function name does not give much information about what it does.
Also, I don't see a compelling reason to use named returns here.
```suggestion
func getSubDomainDNSRecords(client *route53.Route53, hostedZone *route53.HostedZone, dottedClusterDomain string) ([]string, error) {
```